asn1/CommonLedgerProtocol.asn

```diff
@@ -0,0 +1,103 @@
+CommonLedgerProtocol
+DEFINITIONS
+AUTOMATIC TAGS ::=
+BEGIN
+
+IMPORTS
+    UInt8,
+    UInt32,
+    UInt64,
+    UInt128,
+    UInt256
+    FROM GenericTypes
+
+    Timestamp
+    FROM InterledgerTypes
+
+    InterledgerProtocolError
+    FROM InterledgerProtocol
+;
+
+-- Zero or more protocolData entries, each with a unique
+-- protocolId, describe protocol-specific information about
+-- unpaid, paid, and conditionally paid requests and results
+-- sent over the CommonLedgerProtocol.
+-- The protocolId of each entry describes the protocol for
+-- interpreting the data. Peers need to establish beforehand
+-- which protocols each understand, and which numbers they
+-- use, which can be agreed ad-hoc,except that protocolId 1
+-- is reserved for the Interledger protocol.
+-- When multiple protocolData items are sent in a request,
+-- only the first one will be decisive in triggering the
+-- response, the rest of the protocolData items are considered
+-- side protocols that piggyback on the first one.
+protocolData ::= SEQUENCE OF SEQUENCE {
+  protocolId UInt8 UNIQUE,
+  data OCTET STRING
+}
+
+-- Ack, Response, and Error are the response types.
+-- When using these in a CLP packet, the requestId should match
+-- the requestId of the request they respond to.
+
+Ack ::= ProtocolData
```

Comment:
Whoops, Adrian commented "Why can't this be `NULL`?" - I tried to respond, then wanted to cancel responding and accidentally deleted Adrian's comment. ^^

Anyway, my comment is: I can't say that you would *never* want to include data with an ack and setting this to ProtocolData costs one byte and enables lots of use cases in the future. Maybe I want to send a balance update with the ack or the local timestamp when the message was received, or performance data like "it took 8.2ms to process this request" etc.